tfjs-layers/src/layers/preprocessing/preprocessing_utils.ts

```diff
@@ -0,0 +1,84 @@
+import { Tensor, denseBincount, Tensor1D, Tensor2D, TensorLike, mul} from '@tensorflow/tfjs-core';
+import { getExactlyOneTensor } from '../../utils/types_utils';
+import { expandDims} from '@tensorflow/tfjs-core';
+import { ValueError } from '../../errors';
+import * as K from '../../backend/tfjs_backend';
+
+export const int = 'int';
+export const oneHot = 'oneHot';
+export const multiHot = 'multiHot';
+export const count = 'count';
+export const tfIdf = 'tfIdf';
+
+export function encodeCategoricalInputs(inputs: Tensor|Tensor[],
+                                        outputMode: string,
+                                        depth: number,
+                                        weights?: Tensor1D|Tensor2D|TensorLike):
+                                        Tensor|Tensor[] {
+
+  inputs = getExactlyOneTensor(inputs);
+
+  if(inputs.dtype !== 'int32') {
+        inputs = K.cast(inputs, 'int32');
+    }
+
+  if(outputMode === int) {
+    return inputs;
+  }
+
+  const originalShape = inputs.shape;
+
+  if(inputs.rank === 0) {
+    inputs = expandDims(inputs, -1);
+  }
+
+  if(outputMode === oneHot) {
+    if(inputs.shape[inputs.shape.length - 1] !== 1) {
+      inputs = expandDims(inputs, -1);
+    }
+  }
+
+  if(inputs.rank > 2) {
+    throw new ValueError(`When outputMode is not 'int', maximum output rank is 2
+    Received outputMode ${outputMode} and input shape ${originalShape}
+    which would result in output rank ${inputs.rank}.`);
+  }
+
+  const binaryOutput = [multiHot, oneHot].includes(outputMode);
+
+  let denseBincountInput;
+
+  if(inputs.rank === 1) {
+    denseBincountInput = inputs as Tensor1D;
+  }
+
+  if(inputs.rank === 2) {
+    denseBincountInput = inputs as Tensor2D;
+  }
+
+  let binCounts;
```

Comment:
Give this a type. I think it's this, but not sure.
```suggestion
  let binCounts: Tensor1D | Tensor2D;
```